pkg/deploy/service_accounts.go

```diff
@@ -49,43 +56,26 @@ func (s *ServiceAccounts) fetchServiceAccount(nsName string, saName string) (str
 		return "", fmt.Errorf("Internal inconsistency: Expected service account name to not be empty")
 	}
 
-	sa, err := s.coreClient.CoreV1().ServiceAccounts(nsName).Get(context.Background(), saName, metav1.GetOptions{})
+	tokenMgr := token.NewManager(s.coreClient, s.log)
+
+	t, err := tokenMgr.GetServiceAccountToken(nsName, saName, &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			ExpirationSeconds: getInt64Point(5000), // 5 minutes
+		},
+	})
 	if err != nil {
-		return "", fmt.Errorf("Getting service account: %s", err)
+		return "", fmt.Errorf("Failed to get service account token: %s", err)
 	}
 
-	for _, secretRef := range sa.Secrets {
-		secret, err := s.coreClient.CoreV1().Secrets(nsName).Get(context.Background(), secretRef.Name, metav1.GetOptions{})
-		if err != nil {
-			return "", fmt.Errorf("Getting service account secret: %s", err)
-		}
-
-		if secret.Type != corev1.SecretTypeServiceAccountToken {
-			continue
-		}
-
-		return s.buildKubeconfig(secret)
+	cert, err := os.ReadFile(caCertPath)
```

Comment:
i dont believe cert is changing so we can in theory cache it after reading it once.

Reply:
hmm, I suppose if one ever had to rotate the kube-api cert that all deployments would get restarted anyways? Thus we would re-read from disk in that case.